### PR TITLE
Passthrough auth config doc

### PR DESCRIPTION
**Pull request by saiskee**
# Description

Add docs for passing config to Passthrough auth service from AuthConfig

### CODE REVIEW

File: docs/content/guides/security/auth/extauth/passthrough_auth/_index.md [hunk 1]
```diff
@@ -135,7 +135,7 @@ spec:
         # Address of the grpc auth server to query
         address: example-grpc-auth-service.default.svc.cluster.local:9001
         # Set a connection timeout to external service, default is 5 seconds
-        connectionTimeout: 3s
+        connectionTimeout: 3
```
Comment (sam-heilbron) [inline docs/content/guides/security/auth/extauth/passthrough_auth/_index.md:138]: Do we need to change this value as well: https://github.com/solo-io/gloo/blob/master/docs/examples/grpc-passthrough-auth/authconfig.yaml#L13?

Comment (saiskee) [reply docs/content/guides/security/auth/extauth/passthrough_auth/_index.md:138]: good catch, this was a mistaken deletion.

File: docs/content/guides/security/auth/extauth/passthrough_auth/_index.md [hunk 2]
```diff
@@ -223,6 +223,36 @@ State from other auth steps is sent to the passthrough service via [CheckRequest
...
+    name: passthrough-auth
+    namespace: gloo-system
+  spec:
+    configs:
+    - passThroughAuth:
+        grpc:
+        # Address of the grpc auth server to query
+        address: example-grpc-auth-service.default.svc.cluster.local:9001
+        # Set a connection timeout to external service, default is 5 seconds
+        connectionTimeout: 3s
```
Comment (sam-heilbron) [inline docs/content/guides/security/auth/extauth/passthrough_auth/_index.md:247]: I think these need to be indented, they are inline with grpc right now

File: docs/content/guides/security/auth/extauth/passthrough_auth/_index.md [hunk 3]
```diff
@@ -223,6 +223,36 @@ State from other auth steps is sent to the passthrough service via [CheckRequest
...
 State from the passthrough service can be sent to other auth steps via [CheckResponse DynamicMetadata](https://github.com/envoyproxy/envoy/blob/50e722cbb0486268c128b0f1d0ef76217387799f/api/envoy/service/auth/v3/external_auth.proto#L126) under a unique key: `solo.auth.passthrough`.
 
+### Passing in custom configuration to Passthrough Auth Service from AuthConfigs
+{{% notice note %}}
+This feature was introduced with **Gloo Edge Enterprise**, release 1.6.15. If you are using an earlier version, this will not work.
+{{% /notice %}}
+
+Custom config can be passed from gloo to the passthrough authentication service. This can be achieved using the `config` field under Passthrough Auth in the AuthConfig:
+
+{{< highlight shell "hl_lines=9-14" >}}
```
Comment (sam-heilbron) [inline docs/content/guides/security/auth/extauth/passthrough_auth/_index.md:233]: Which lines are you looking to highlight? I don't think these will include the config block right now

File: docs/content/guides/security/auth/extauth/passthrough_auth/_index.md [hunk 4]
```diff
@@ -223,6 +223,36 @@ State from other auth steps is sent to the passthrough service via [CheckRequest
...
+        address: example-grpc-auth-service.default.svc.cluster.local:9001
+        # Set a connection timeout to external service, default is 5 seconds
+        connectionTimeout: 3s
+      config:
+        customKey1: "customConfigStringValue"
+        customKey2: false
+EOF
+{{< /highlight >}}
+
+This config is accessible via the [CheckRequest FilterMetadata](https://github.com/envoyproxy/envoy/blob/50e722cbb0486268c128b0f1d0ef76217387799f/api/envoy/service/auth/v3/external_auth.proto#L36) under a unique key: `solo.auth.passthrough.config`.
```
Comment (sam-heilbron) [inline docs/content/guides/security/auth/extauth/passthrough_auth/_index.md:254]: 🎉

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**